packages/ckeditor5-ui/src/dropdown/menu/definition/dropdownmenudefinitiontypings.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/dropdown/menu/definition/dropdownmenudefinitiontypings
+ */
+
+import type { DropdownMenuOrFlatItemView } from '../typings.js';
+
+/**
+ * Represents a child definition of a dropdown menu.
+ * It can be either a `DropdownMenuDefinition` or a `DropdownMenuOrFlatItemView`.
+ */
+export type DropdownMenuChildDefinition = DropdownMenuDefinition | DropdownMenuOrFlatItemView;
```

Comment:
> I don't like the order of things in this file.

Reordered, definition object is reduced.

> Do we have an actual requirement where we must pass instances

At least two reasons:

1. In some of our plugins, like in [AI](https://github.com/cksource/ckeditor5-commercial/blob/master/packages/ckeditor5-ai/src/aiassistantui.ts#L576) there is filterable view. From what I can see, it used to operate on plain instances that used to be a bit customized (at least manually created). They usually inherit from `ListItemView` to store some additional information about commands that are later used to control `isOn` state right [there](https://github.com/cksource/ckeditor5-commercial/blob/master/packages/ckeditor5-ai/src/ui/dropdown/aicommandslistview.ts#L47). The other thing is that it also watches on `execute` on such items like [here](https://github.com/cksource/ckeditor5-commercial/blob/master/packages/ckeditor5-ai/src/aiassistantui.ts#L255C24-L255C31). 
2. It simplifies binding `isOn` / `isEnabled` states in filter view. Let's assume we are filtering for `Butter`, it displays found menu items that `isEnabled` state is bound to these ancestor menu item instances. It would be not possible in JSON approach, where the listener should be implemented for both filtering view and menu view.

Reply:
After discussion, we decided I'll keep possibility to `View` to dropdown menu and add additional flat item definition entry with `onExecute`.